### PR TITLE
feat: replace placeholder image with repo social card

### LINKED ISSUE #17: Feature: Replace placeholder image with repo social card
### Type of feature

Feature

### Current behavior

Prior to #13, this app was pulling in the social cards from GitHub. 

https://github.com/open-sauced/hot-sauce/blob/main/src/components/SocialImage.jsx

This is the previous fetching of the social card image. 

https://github.com/open-sauced/hot-sauce/blob/c3e28250cc27ee7b43b5134b1e883bc8c2864338/src/pages/index.astro#L14-L21

Previous PR where this was working https://github.com/open-sauced/hot-sauce/pull/1/files

### Desired solution

Add the social image in replace of the placeholder. See the previous PR where this was:
Screen Shot 2021-12-17 at 4 55 20 PM

Example pulling images from open graph.

Group 2

### Additional context

[Previous site with social card
](https://4000735cf7f--hot-sauced-ui.netlify.app)
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hokagedemehin**
## What type of PR is this? (check all applicable)

- [x] ✨ Feature

## Description

This PR replaces the placeholder holder images on each card with the right repo social card. After so much research, I was able to work out a code that works. Also, I made some little corrections to the tailwind CSS typo errors and reduced the size of the text of the action buttons at the bottom of each card so as to correct the overflow problem. I still believe some corrections can be made on each card for uniformity.

## Related Tickets & Documents

Fixes #17 

## Added tests?

- [x] 🙅 no, because they aren't needed

### CODE REVIEW

**Review by kelvinsekx — CHANGES_REQUESTED** [review]
Ok. I saw the corrections already

**Review by kelvinsekx — CHANGES_REQUESTED** [review]
Can you please add a minHeight to the social card container or any other trick to the Action Buttons container just to make it fixed to the bottom?

I will try to share an image to show you want I intend better.

**Comment by kelvinsekx** [comment]
Screenshot_20211227-104051~2

**Comment by kelvinsekx** [comment]
Screenshot_20211227-104038~2

**Comment by hokagedemehin** [comment]
I actually thought of the action button sticking to the bottom and not floating everywhere. I will work on that.

Also for the repo social card, I will work on it showing full on all possible screens

**Comment by hokagedemehin** [comment]
hello @kelvinsekx 

Could you take a look at the checks for this PR, the codeQL and Semantics checks seems to be taking a very long time to complete

**Comment by 0-vortex** [comment]
> # PR Compliance Checks
> Thank you for your Pull Request! We have run several checks on this pull request in order to make sure it's suitable for merging into this project. The results are listed in the following section.
> 
> ## Watched Files
> This pull request modifies specific files that require careful review by the maintainers.
> 
> Files Matched
> 
> * npm-shrinkwrap.json

The changes here are good, it's just optional scripts for more esbuild platforms!

**Comment by hokagedemehin** [comment]
> > # PR Compliance Checks
> > Thank you for your Pull Request! We have run several checks on this pull request in order to make sure it's suitable for merging into this project. The results are listed in the following section.
> > ## Watched Files
> > This pull request modifies specific files that require careful review by the maintainers.
> > Files Matched
> > 
> > * npm-shrinkwrap.json
> 
> The changes here are good, it's just optional scripts for more esbuild platforms!

Please How do I install packages in this project??

**Comment by 0-vortex** [comment]
`npm install --save "packageName"@"^X.Y.Z"` for production, `npm install --save-dev "packageName"@"^X.Y.Z"` for development - are you having troubel installing a specific package?

**Comment by hokagedemehin** [comment]
> `npm install --save "packageName"@"^X.Y.Z"` for production, `npm install --save-dev "packageName"@"^X.Y.Z"` for development - are you having troubel installing a specific package?

Yes, I wanted to install react tooltip, but I was getting this error

"module require was not found" inside one of the node_module files

**Comment by 0-vortex** [comment]
> > `npm install --save "packageName"@"^X.Y.Z"` for production, `npm install --save-dev "packageName"@"^X.Y.Z"` for development - are you having troubel installing a specific package?
> 
> Yes, I wanted to install react tooltip, but I was getting this error
> 
> "module require was not found" inside one of the node_module files

That's not related to the actual dependency but the way you require it.

Change:

```jsx
const packageName = require('package-name');
```

to:

```jsx
import packageName from 'package-name';
```

Can help quicker if you can upload the failing code to a branch or link the dependency itself 🍕

File: src/components/PostGrid.jsx [hunk 1]
```diff
@@ -14,62 +13,88 @@ function PostGrid({ data }) {
...
+        {/* Cover photo */}
+        <div className="w-full bg-transparent h-28 overflow-hidden rounded-md mb-2 flex justify-center">
           <img
-            className="object-cover"
-            src={getAvatar(data?.contributors[1])}
-            alt="Avatar 02"
-            width={500}
-            height={500}
+            className="object-contain h-full"
+            src={`https://opengraph.githubassets.com/1/${data.repo_name}`}
```
Comment (bdougie) [inline src/components/PostGrid.jsx:61]: This is perfect

**Review by bdougie — COMMENTED** [review]
The PostGrif image seems redundant. I think it would be best remove the stars and issues favicons and lean on the image for that data. 

Also think the upvote can be pinned to the top for the grid to allow a larger image.

**Comment by hokagedemehin** [comment]
> The PostGrif image seems redundant. I think it would be best remove the stars and issues favicons and lean on the image for that data. 
> 
> Also think the upvote can be pinned to the top for the grid to allow a larger image.

The image used in the PostGrid is from a link I used in the code. That is what GitHub renders on twitter and any social media platform, i think. Is that the image you are talking about?

**Comment by bdougie** [comment]
Yes that's the image. It already provides stars and issue details in the rendered issue.

File: src/components/PostGrid.jsx (hunk 1 above)
Comment (hokagedemehin) [reply src/components/PostGrid.jsx:61]: I will remove the stars and issues favicons and also put the upvote button at the top right corner of the card

**Comment by hokagedemehin** [comment]
I will close this Pull Request and start again. hopefully it should not fail any test

**Pull request closed without merging**